### ISSUE TITLE
process killed in Warp but not Terminal

### ISSUE THREAD

**Issue by meshula** [warpdotdev/Warp#3448]
### Describe the bug

A program I compiled myself, that uses PySide, is killed when launching under Warp, but not Terminal. Launching within terminal via lldb works.

### To reproduce

1. `git clone https://github.com/PixarAnimationStudios/OpenUSD.git`
2. `python3 /path/to/OpenUSD/build_scripts/build_usd.py --prefer-speed-over-safety --python --usd-imaging --openimageio --openvdb --alembic  --embree --materialx --generator Xcode /var/tmp/usd0727/`
3. `export PATH=/var/tmp/usd0727/bin:$PATH`
4. `export PYTHONPATH=/var/tmp/usd0727/lib/python`
5. `usdview` - observe some help text is printed
6. from xcode, open `/var/tmp/usd0727/build/OpenUSD/usd.xcodeproj`, pick `install` target, and build
7. return to Warp
8. `usdview` - observe message `zsh: killed     usdview`
9. open Terminal, repeat step 3, 4, 5, observe help text

### Expected behavior

`zsh: killed     usdview` should not occur if it does not occur in `Terminal`.

### Screenshots

<img width="1317" alt="image" src="https://github.com/warpdotdev/Warp/assets/501238/432fe645-11d4-4461-83eb-79a6157d744e">

### Operating system

MacOS

### Operating system and version

13.4.1

### Shell Version

zsh 5.9 (x86_64-apple-darwin22.0)

### Current Warp version

v0.2023.07.25.08.03.stable_00

### Regression

Yes, this bug started recently or with an X Warp version

### Recent working Warp date

first started happening in May of 2023

### Additional context

workaround: run my compiled software in Terminal.

### Does this block you from using Warp daily?

No

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b8107fdf-ba31-488d-b103-d271c89cac3e

None

**Comment by meshula** [comment]
Thank you for taking a look. I can appreciate building our project has a learning curve, I am hoping to narrow the symptoms to something much smaller and more debuggable. I will update when I have more information.

**Comment by dannyneira** [comment]
Any luck with narrowing down the issue?

**Comment by meshula** [comment]
Hi, thank you for following up! I did narrow the problem down specifically to a non-Warp related cause. I found a step in my project's build process that forced a codesigning step on a dylib installed by pip. There seems to be an aspect of signing the same project from different processes, and not being absolutely thorough about it, that triggers the process kill.

This is definitely not a Warp issue; the fact that it happened in Warp was merely a symptom of a complex and not-yet fully correct build process.
